src/fhirtypes/common.ts

```diff
@@ -198,6 +198,20 @@ export function setPropertyOnInstance(
           if (current[key] == null) current[key] = {};
           current = current[key];
         } else {
+          // Check if the instance  already has the element being defined
+          if (current.hasOwnProperty([pathPart.base])) {
+            // Check if the instance already has a quantity element
+            // Quantity elements are the only FHIR types with both a code and value
+            if (current[pathPart.base].hasOwnProperty('value')) {
+              if (assignedValue.hasOwnProperty('code')) {
```

Comment:
The conditions in the three if statements can safely be combined into a single if statement, since the only code that modifies data is nested inside all three.